tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php

```diff
@@ -0,0 +1,282 @@
+<?php
+namespace Elementor\Tests\Phpunit\Elementor\Core\Common\Modules\Connect\Apps;
+
+use Elementor\Testing\Elementor_Test_Base;
+use Elementor\Core\Common\Modules\Connect\Apps\Base_App;
+use Elementor\Tests\Phpunit\Elementor\Core\Common\Modules\Connect\Apps\Mock\Mock_App;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Test_Common_App extends Elementor_Test_Base {
+	private $http_stub;
+
+	/**
+	 * @var Mock_App
+	 */
+	private $app_stub;
+
+	public function setUp() {
+		parent::setUp();
+
+		require_once __DIR__ . '/mock/mock-app.php';
+
+		$this->http_stub = $this->getMockBuilder( \WP_Http::class )
+			->setMethods( [ 'request' ] )
+			->getMock();
+
+		$this->app_stub = new Mock_App();
+		$this->app_stub->set_http( $this->http_stub );
+	}
+
+	public function test_http_request() {
+		// Arrange
+		$endpoint = 'test/1';
+
+		$this->http_stub->method( 'request' )
+			->with(
+				$this->equalTo( Mock_App::BASE_URL . '/' . $endpoint ),
+				$this->equalTo( [
+					'headers' => [],
+					'method' => 'POST',
+					'timeout' => 25,
+					'body' => [
+						'a' => 1,
+					],
+				] )
+			)
+			->willReturn( [
+				'response' => [
+					'code' => 200,
+				],
+				'body' => '{"status": "success"}',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', $endpoint, [
+			'body' => [
+				'a' => 1,
+			]
+		] );
+
+		// Assert
+		$this->assertEquals( (object) [ 'status' => 'success' ], $result );
+	}
+
+	public function test_http_request__return_error_when_not_response_code() {
+		// Arrange
+		$this->http_stub->method( 'request' )
+			->willReturn( [
+			    'response' => [
+			        'code' => 0,
+			    ],
+			    'body' => '{"status": "success"}',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test' );
+
+		// Assert
+		$this->assertInstanceOf( \WP_Error::class, $result );
+	}
+
+	public function test_http_request__server_sent_success_without_body() {
+		// Arrange
+		$this->http_stub->method( 'request' )
+			->willReturn( [
+			    'response' => [
+			        'code' => 200,
+			    ],
+			    'body' => 'null',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test' );
+
+		// Assert
+		$this->assertEquals( 1, $result );
```

Comment:
Who returns the `1`?

Reply:
When the body is a string with the value of 'null' and the code change the value to `true`, then after the json_decode it becomes `1`. this is the behavior that was before this change.

Reply:
Explain in code.